lib/modules/datasource/util.ts

```diff
@@ -8,3 +11,30 @@ export function isArtifactoryServer<T = unknown>(
 ): boolean {
   return is.string(res?.headers[JFROG_ARTIFACTORY_RES_HEADER]);
 }
+
+export async function getGoogleAuthToken(): Promise<string | undefined> {
```

Comment:
This doesn't look correct to me: Even if no `accessToken` was retrieved, a dummy `oauth2accesstoken:undefined` is constructed. Also wrapping `${'oauth2accesstoken'}` in template literals is unnecessary, as it is a constant string. I'd also suggest returning `null` instead of `undefined`, since your intention is to signal the lake of a value. 

So overall like this:

```suggestion
export async function getGoogleAuthToken(): Promise<string | null> {
  try {
    const googleAuth: GoogleAuth = new GoogleAuth({
      scopes: 'https://www.googleapis.com/auth/cloud-platform',
    });
    const accessToken = await googleAuth.getAccessToken();
    if (accessToken) {
      // sanitize token
      addSecretForSanitizing(accessToken);
      return Buffer.from(`oauth2accesstoken:${accessToken}`).toString('base64');
    } else {
      logger.warn(
        'Could not retrieve access token using google-auth-library getAccessToken'
      );
    }
  } catch (err) {
    if (err.message?.includes('Could not load the default credentials')) {
      return null;
    } else {
      throw err;
    }
  }

  return null;
}
```

Reply:
You are correct 😮. Good catch! Thx